test/functional/test-xhr.js

```diff
@@ -268,15 +268,13 @@ describe('XHR', function() {
               });
         });
 
-        it('should not return a resolved indented promise', () => {
+        it('should not resolve after rejecting promise', () => {
           mockXhr.status = 500;
           mockXhr.responseText = '{"a": "hello"}';
           mockXhr.headers['Content-Type'] = 'application/json';
           mockXhr.getResponseHeader = () => 'application/json';
-          const promise = assertSuccess(
-              createResponseInstance('{"a": 2}', mockXhr));
-          promise.should.be.rejected;
```

Comment:
Yeah that was previously, we added that test to catch that case and then removed the return statement. 

> We did immediately resolve when there was a JSON error response, which we definitely should have caught.

Yes the edit to this test makes sure we catch that.